src/engine/enginemaster.cpp

```diff
@@ -97,6 +100,13 @@ EngineMaster::EngineMaster(UserSettingsPointer pConfig,
 
     m_pMasterDelay = new EngineDelay(group, ConfigKey(group, "delay"));
     m_pHeadDelay = new EngineDelay(group, ConfigKey(group, "headDelay"));
```

Comment:
> How can the user scratch on beat without listening to the results on headphone?
We can assume that the user listens to the master mix on headphone expects that if he does not hear double beats, the crowd on the does not hear it as well.

Whether the user is listening to the scratched signal in headphones, booth, or master output is irrelevant. Again, you are confusing the total output delays with the input latency compensation delays. They are applied at different parts of the signal chains and have separate use cases. The general signal path is:

```
channel processing --> input latency compensation --> output delay --> buffer sent to sound card
                                                  /
                                   mic input ----
```

The input latency compensation delays are only for aligning Mixxx's output with live input signals. The output delays are for aligning what the DJ hears from master/booth/headphone outputs to avoid hearing double beats.